Makefile.tools

```diff
@@ -0,0 +1,289 @@
+# Pretty much everything needs a newer GCC than will be availiable on any CAD
+# tools machines.
+$(OBJ_TOOLS_BIN_GCC) $(OBJ_TOOLS_BIN_GXX): $(OBJ_TOOLS_BIN_DIR)/gcc-$(GCC_VERSION)/stamp
+	touch $@
+
+$(OBJ_TOOLS_BIN_DIR)/gcc-$(GCC_VERSION)/stamp: $(OBJ_TOOLS_SRC_DIR)/gcc-$(GCC_VERSION)/build/Makefile
+	+MAKEFLAGS="$(TOOLS_MAKEFLAGS)" $(SCHEDULER_CMD) -- src/tools/build-gcc --srcdir $(dir $<) --bindir $(dir $@) --logfile $(dir $<)/build.log
+	@date > $@
+
+$(OBJ_TOOLS_SRC_DIR)/gcc-$(GCC_VERSION)/build/Makefile: $(OBJ_TOOLS_SRC_DIR)/gcc-$(GCC_VERSION)/configure
+	@mkdir -p $(dir $@)
+	cd $(dir $@); ../configure --prefix=$(abspath $(OBJ_TOOLS_BIN_DIR)/gcc-$(GCC_VERSION)) --enable-languages=c,c++ --disable-multilib --disable-checking
+
+$(OBJ_TOOLS_SRC_DIR)/gcc-$(GCC_VERSION)/configure: $(PLSI_CACHE_DIR)/distfiles/gcc-$(GCC_VERSION).tar.gz
+	@rm -rf $(dir $@)
+	@mkdir -p $(dir $@)
+	tar -xpf $< --strip-components=1 -C $(dir $@)
+	touch $@
+
+$(PLSI_CACHE_DIR)/distfiles/gcc-$(GCC_VERSION).tar.gz:
+	@mkdir -p $(dir $@)
+	wget https://ftp.gnu.org/gnu/gcc/gcc-$(GCC_VERSION)/gcc-$(GCC_VERSION).tar.gz -O $@
+
+##############################################################################
+# pconfigure and related tools
+##############################################################################
+
+# _plsi_ contrib pconfigure targets = OBJ_TOOLS_BIN_PTEST OBJ_TOOLS_BIN_PCONFIGURE OBJ_TOOLS_BIN_PPKGCONFIG OBJ_TOOLS_BIN_PHC
+
+OBJ_TOOLS_BIN_PTEST = $(OBJ_TOOLS_BIN_DIR)/pconfigure/bin/ptest
+OBJ_TOOLS_BIN_PCONFIGURE = $(OBJ_TOOLS_BIN_DIR)/pconfigure/bin/pconfigure
+OBJ_TOOLS_BIN_PPKGCONFIG = $(OBJ_TOOLS_BIN_DIR)/pconfigure/bin/ppkg-config
+OBJ_TOOLS_BIN_PHC = $(OBJ_TOOLS_BIN_DIR)/pconfigure/bin/phc
+
+$(OBJ_TOOLS_BIN_PTEST) \
+$(OBJ_TOOLS_BIN_PCONFIGURE) \
+$(OBJ_TOOLS_BIN_PPKGCONFIG) \
+$(OBJ_TOOLS_BIN_PHC): $(OBJ_TOOLS_BIN_DIR)/pconfigure/stamp
+	touch $@
+
+$(OBJ_TOOLS_BIN_DIR)/pconfigure/stamp: $(OBJ_TOOLS_SRC_DIR)/pconfigure/Makefile
+	MAKEFLAGS="$(TOOLS_MAKEFLAGS)" $(MAKE) -C $(OBJ_TOOLS_SRC_DIR)/pconfigure CC=$(abspath $(CMD_GCC)) CXX=$(abspath $(CMD_GXX)) install
+	@date > $@
+
+$(OBJ_TOOLS_SRC_DIR)/pconfigure/Makefile: $(OBJ_TOOLS_SRC_DIR)/pconfigure/stamp $(CMD_GXX)
+	@mkdir -p $(dir $@)
+	mkdir -p $(OBJ_TOOLS_SRC_DIR)/pconfigure/Configfiles
+	rm -f $(OBJ_TOOLS_SRC_DIR)/pconfigure/Configfiles/local
+	echo 'PREFIX = $(abspath $(OBJ_TOOLS_BIN_DIR))/pconfigure' >> $(OBJ_TOOLS_SRC_DIR)/pconfigure/Configfiles/local
+	echo 'LANGUAGES += c++' >> $(OBJ_TOOLS_SRC_DIR)/pconfigure/Configfiles/local
+	echo 'LINKOPTS += -L,$(abspath $(CMD_PATH_GCC_LIB64))' >> $(OBJ_TOOLS_SRC_DIR)/pconfigure/Configfiles/local
+	echo 'LINKOPTS += -Wl,-rpath,$(abspath $(CMD_PATH_GCC_LIB64))' >> $(OBJ_TOOLS_SRC_DIR)/pconfigure/Configfiles/local
+	+cd $(OBJ_TOOLS_SRC_DIR)/pconfigure; $(SCHEDULER_CMD) --max-threads=1 -- ./bootstrap.sh --cc $(abspath $(CMD_GCC)) --cxx $(abspath $(CMD_GXX)) --cxxflags "-L,$(abspath $(CMD_PATH_GCC_LIB64)) -Wl,-rpath,$(abspath $(CMD_PATH_GCC_LIB64))"
```

Comment:
That is an artefact of pconfigure and since we're replacing pcad very soon anyway, I'm not sure it's worth the time to investigate, unless you feel otherwise?